mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -99,16 +98,15 @@ class ConvConverter final
   }
 
   LogicalResult
-  matchAndRewrite(migraphx::ConvolutionOp op, OpAdaptor adaptor,
-                  ConversionPatternRewriter &rewriter) const final {
-    auto operands = adaptor.getOperands();
+  matchAndRewrite(T op, typename OpConversionPattern<T>::OpAdaptor adaptor,
```

Comment:
first, I agree with your attempt to avoiding code duplication.

Please put a comment this will be used only with a convolution and possibly with a todo to use a base class (i.e. an interface) as it becomes hard to read what this conversion pattern is converting now.

(Im fine with this for now but however it would be better to use MIGraphXConvInterface rather than a generic template -- if you agree, lets put a todo with a ticket do the interface later.)

Reply:
Good point, I think I can change T to be ConvType and add a comment now to indicate what this conversion pattern targets at.

As far as the quant_convolution, I don't think we need to generalize it more: The tablegen type field captured the difference between the two ops. It is fairly straightforward to tell for the existing implementation.

Giving a comparison with what we did we rock conv2d ops, we have both practice available:
 - In Conv2DRewritePattern, it is using a templatized manner to do it, very similar with here. This is because we need to materialize each instance of class for that particular lowering type. 
 - In MatchLayoutsToInput pattern, it is using `RockConvInterface`. In this creating class instance copies would be create duplicated compilation unit therefore we should be using a common interface to address it.

I'm hoping my generalization of the two cases are right. Now with that assumption, in the quant_convolution case we indeed need to materialize the two independent lowering types. Therefore it is reasonable to pick the first practice. I'd agree with us creating a ticket only when the second scenario exists: That there comes a use case where a lowering pipeline to handle this group of conv ops.